Create BoxedKernel as a subset of KernelFunction

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #67096
* __->__ #67095
* #67094
* #67093

KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

peterbell10 committed:
Remove IValue dependency from KernelFunction.h

KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` and the `makeFromUnboxed` constructors. Moving
these into free functions in another header allows making boxed kernel
functions without the need to include IValue (see next PR in the
stack).

This does make the interface more awkward, but I think it's worthwhile
since it's an internal API that's only used in a couple of places.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove IValue dependency from KernelFunction.h"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` and the `makeFromUnboxed` constructors. Moving
these into free functions in another header allows making boxed kernel
functions without the need to include IValue (see next PR in the
stack).

This does make the interface more awkward, but I think it's worthwhile
since it's an internal API that's only used in a couple of places.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove IValue dependency from KernelFunction.h"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` and the `makeFromUnboxed` constructors. Moving
these into free functions in another header allows making boxed kernel
functions without the need to include IValue (see next PR in the
stack).

This does make the interface more awkward, but I think it's worthwhile
since it's an internal API that's only used in a couple of places.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove IValue dependency from KernelFunction.h"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` and the `makeFromUnboxed` constructors. Moving
these into free functions in another header allows making boxed kernel
functions without the need to include IValue (see next PR in the
stack).

This does make the interface more awkward, but I think it's worthwhile
since it's an internal API that's only used in a couple of places.

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289813).

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289813).

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289813).

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Update on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Rebase after master CI break on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289813).

peterbell10 committed:
Only build IValueInterface when necessary on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289813).

peterbell10 committed:
Rebase and fix merge conflicts on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]

peterbell10 committed:
Rebase on "Create BoxedKernel as a subset of KernelFunction"


KernelFunction.h includes ivalue.h (and so depends on Tensor.h)
exclusively for `call` (unboxed) and the `makeFromUnboxed`
constructors. Moving the boxed-only components into a seperate
BoxedKernel function allows boxed kernels to be registered without
including `ivalue.h` (see next PR in the stack).

Differential Revision: [D32289813](https://our.internmc.facebook.com/intern/diff/D32289813)

[ghstack-poisoned]
